### ISSUE TITLE
Recognize all plumber files

### ISSUE THREAD

**Issue by blairj09** [rstudio/rstudio#8118]
When a "traditional" Plumber file is being edited:

```r
library(plumber)

#* @apiTitle Plumber Example API

#* Echo back the input
#* @param msg The message to echo
#* @get /echo
function(msg = "") {
    list(msg = paste0("The message is: '", msg, "'"))
}
```

the IDE presents the option to Run API:
![image](https://user-images.githubusercontent.com/10444878/96547566-907e6280-1269-11eb-8afc-81f032700bb3.png)

However, in version 1.0.0, Plumber now includes a "tidy interface", so APIs can also be defined as follows:
```r
library(plumber)

pr() %>% 
  pr_get(path = "/echo",
         handler = function(msg = "") {
           list(msg = paste0("The message is: '", msg, "'"))
         })
```

It would be nice if the IDE presented the same Run API option when editing these files as well

UPDATE (2021-01-05)
Per comments later in this issue, the recommended best practice we should support is:

```r
library(plumber)

#* @plumber
function(pr) {
  pr %>%
    pr_get(path = "/echo",
           handler = function(msg = "") {
             list(msg = paste0("The message is: '", msg, "'"))
           })
}
```

**Comment by kevinushey** [comment]
This is where we detect plumber file types:

https://github.com/rstudio/rstudio/blob/e527463081fc095bbe69d43ae2949ff03f44e9a5/src/cpp/session/modules/plumber/SessionPlumber.cpp#L44-L53

Is there some sort of "dummy" Plumber-style roxygen comment that we could advocate using even with the "tidy" style?

**Comment by jmcphers** [comment]
This looks like it's going to be tricky to do correctly with a regex, and is a minor enhancement that does not meet the bar for 1.4.

**Comment by blairj09** [comment]
I agree that finding the correct recipe here is likely going to be challenging. I'll work with @schloerke to see if we can think of possible solutions for correctly identifying "plumbable" files.

**Comment by kevinushey** [comment]
A "simple" solution would be to require, or advocate for, plumber files to have some sort of plumber-style "header"; e.g.

```
#* apiTitle This is my Plumber API!
```

Then RStudio could easily detect that via a regex.

**Comment by blairj09** [comment]
After discussing with @schloerke, I think we can just expand the current regex list to include new tags that have been introduced:
```
R"(^#\*\s*@(get|put|post|filter|assets|use|delete|head|options|patch|plumber|serializer|parser|preempt|response|tag|apiTitle|apiDescription|apiTOS|apiContact|apiLicense|apiVersion|apiTag)\s)");
```

This won't help in the case I originally described, but we've decided best practice is something like the following:
```r
library(plumber)

#* @plumber
function(pr) {
  pr %>%
    pr_get(...)
}
```

Which would work with the expanded set of fields I suggested.

**Comment by blairj09** [comment]
It would be great to see this implemented. Changes in Plumber mean it's more likely users will start creating APIs that don't match the current regex and will therefore lose the convenience offered by the Run API button.

**Comment by gtritchie** [comment]
Fixed in juliet-rose via #8686.

**Comment by valerie-rstudio** [comment]
### Verified Fixed

    RStudio Edition : Desktop
    RStudio Version : 1.4.1533
    OS Version      : mac OS 11.1
    R Version       : 4.0.2

Verified that when the new plumber API tag `#* @plumber` is included in an R script, the Run API button appears and functions as expected.
